node/src/accountant/mod.rs

```diff
@@ -2773,15 +3251,22 @@ mod tests {
             .consuming_wallet(make_wallet("consuming"))
             .pending_payable_daos(vec![ForPendingPayableScanner(pending_payable_dao)])
             .build();
-        subject.request_transaction_receipts_sub_opt = Some(make_recorder().0.start().recipient());
+        let system = System::new("test");
+        let (blockchain_bridge, _, blockchain_bridge_recording_arc) = make_recorder();
+        let blockchain_bridge_addr = blockchain_bridge.start();
+        subject.request_transaction_receipts_sub_opt = Some(blockchain_bridge_addr.recipient());
         let flag_before = subject.scanners.initial_pending_payable_scan();
 
         let hint = subject.handle_request_of_scan_for_pending_payable(None);
 
+        System::current().stop();
+        system.run();
         let flag_after = subject.scanners.initial_pending_payable_scan();
-        assert_eq!(hint, SchedulingAfterHaltedScan::DoNotSchedule);
+        assert_eq!(hint, ScanRescheduleAfterEarlyStop::DoNotSchedule);
         assert_eq!(flag_before, true);
         assert_eq!(flag_after, false);
+        let blockchain_bridge_recording = blockchain_bridge_recording_arc.lock().unwrap();
+        let _recorded_msg = blockchain_bridge_recording.get_record::<RequestTransactionReceipts>(0);
```

Comment:
Instead of `get_record()`, you may use `get_record_opt`, and perhaps do something like this:

```rust
assert!(blockchain_bridge_recording.get_record_opt::<RequestTransactionReceipts>(0).is_some());
```

Reply:
It is alone a strong enough assertion. You cannot pass it falsely unless you panic.

Reply:
Haha. I accidentally replied twice on the same subject. Well, three times it is by now. :)